src/Facility.CodeGen.CSharp/CSharpGenerator.cs

```diff
@@ -536,6 +536,22 @@ private CodeGenFile GenerateDto(ServiceDtoInfo dtoInfo, Context context)
 							code.WriteLine("return null;");
 						}
 					}
+
+					var fieldInfosForProtobufModel = fieldInfos.Where(x => context.GetFieldType(x).Kind is ServiceTypeKind.Result).ToList();
+					if (fieldInfosForProtobufModel.Count != 0)
+					{
+						code.WriteLine();
+						code.WriteLine($"static {fullDtoName}()");
+						using (code.Block())
+						{
+							foreach (var fieldInfo in fieldInfosForProtobufModel)
+							{
+								// TODO: this won't work when multiple types use the same ServiceResult<T>
```

Comment:
Is the fix a utility method that prevents adding the same type twice?

Reply:
I don't know that this is even the right way to do it. There may be a way to get protobuf-net to do it with a generic type.